demo/admin/src/products/future/generated/ProductsGrid.tsx

```diff
@@ -98,7 +99,10 @@ function ProductsGridToolbar() {
     );
 }
 
-export function ProductsGrid(): React.ReactElement {
+type Props = {
+    baseFilter?: GQLProductFilter;
```

Comment:
How about using preFilter or `prefilter`?

Reply:
IMO prefilter sounds like it's filtered beforehand and not at the same time (which practically would be the same result...). I used baseFilter as it's the base, and all user-added filters are added to this base.

and maybe one could assume prefilter is like preset-filters, and could be changeable via GUI. and that would be a different feature. (but this could probably be achieved with editable: true field per filter-object)

Reply:
why not simply `filter`?

Reply:
`filter` seems to be a good idea, because the gql-var is called the same. so in some future refactor this could be simplified to behave similar to required gql-vars.

Reply:
> why not simply `filter`?

Fine by me

Reply:
@nsams filter is conflicting with filter from graphql-anywhere. how would you solve this in a real world scenario? probably don't name the prop filter?

Reply:
I would prefer to rename the imported filter-function, as mentioned above, it _probably_ simplifies how gql-params could be implemented. (but I assume this is kind of premature optimization, because I should not know what I'm probably going to do in the next pull-request...)

Reply:
> I would prefer to rename the imported filter-function

:+1:

(yes, thinking about the future is a good idea)